### DISCUSSION TITLE
Adding redux-persist or similar functionality to RTK

### DISCUSSION THREAD

**Discussion by seanmcquaid** [reduxjs/redux-toolkit#3332] [Ideas]
It looks like the maintainers of redux-persist are no longer maintaining the package. 

It looks like Zustand provides similar utils out of the box for their users so I think the argument could be made that RTK could include this as well. 

Would love to hear other people's thoughts on this as well!

**Comment by markerikson** [comment]
We've generally said that we're _not_ looking to add persistence or Next integration to RTK, specifically _because_ `redux-persist` and `next-redux-wrapper` already exist.

The other issue is maintenance burden. We barely have enough time to work on RTK and its current feature set as it is (see the giant pile of open "codegen" related issues here in the repo).  We don't have time or capacity to take on maintenance of entire other existing libraries.    (There was actually a similar request about Normalizr a few months ago, and we ended up not doing anything to adopt it either.)

**Reply by seanmcquaid** [reply]
That's absolutely fair! Just an idea. Appreciate all of the work ya'll are doing!

**Comment by phryneas** [comment]
Actively maintained or not - it has a big feature set, allows for plugins, and is pretty much "done". Is there anything about it that's broken for you?

**Reply by seanmcquaid** [reply]
Nothing is necessarily broken, I more just get anxious relying on a package that isn't actively being contributed to when so many changes in the ecosystem are occurring. Not intentionally trying to add more work for ya'll, sorry for raising this!

**Reply by phryneas** [reply]
I honestly wouldn't worry too much - changes in the ecosystem hardly matter to `redux-persist`. 
The only thing would be if we get a successor to `localStorage`, but even then LocalStorage would hang around in browsers for another decade - and you can just always use a different storage adapter, because it is very pluggable.

**Reply by seanmcquaid** [reply]
That's absolutely valid! Thank you for weighing in here, ya'll!
